frontend/src/components/texts/TextEditorCategory.vue

```diff
@@ -0,0 +1,52 @@
+<script lang="ts" setup>
+import { nextTick } from "vue"
+import TextEditor from "../input/TextEditor.vue"
+import CategoryWrapper from "@/components/CategoryWrapper.vue"
+import { TextAreaInputAttributes } from "@/components/input/types"
+
+interface Props {
+  id: string
+  label: string
+  shouldShowButton: boolean
+  modelValue?: string
+  fieldSize?: TextAreaInputAttributes["fieldSize"]
+}
+
+const props = withDefaults(defineProps<Props>(), {
+  modelValue: undefined,
+  fieldSize: "big",
+})
+
+defineEmits<{
+  "update:modelValue": [value: string]
+}>()
+
+async function focusEditor() {
+  await nextTick()
+  const editorElement = document.getElementById(props.id)
+    ?.firstElementChild as HTMLElement
+  editorElement?.focus()
+}
+</script>
+
+<template>
+  <CategoryWrapper
+    :label="label"
+    :should-show-button="shouldShowButton"
+    @toggled="focusEditor"
```

Comment:
Here instead of listening for the @toggled, we can just put everything that is in `focusEditor` inside the `TextEditor` component into an `onMounted` hook